R/layers.R

```diff
@@ -28,131 +28,151 @@
-    }else{check_wfs(wfs)}
-
-    # check layers
-    layers <- match.arg(layers, several.ok = TRUE,
-                       choices = emodnet_get_wfs_info(wfs)$layer_name)
-
-    # check filter vector
-    if(!is.null(cql_filter)){
-        checkmate::testCharacter(cql_filter, min.len = 1)
-        if(length(cql_filter) == 1 & length(layers) > 1){
-            cql_filter <- rep(cql_filter, times = length(layers))
-            usethis::ui_info('{usethis::ui_field("cql_filter")} {usethis::ui_code(cql_filter)} recycled across all layers')
-        }
-        if(checkmate::test_named(cql_filter)){
-            cql_filter <- cql_filter[layers]
-            }
-        checkmate::assert_character(cql_filter, len = length(layers))
-    }else{
-        cql_filter <- rep(NA, times = length(layers))
+
+    if (is.null(wfs)) {
+        wfs <- emodnet_init_wfs_client(service, service_version)
+    }
+
+    check_wfs(wfs)
+
+    # check layers -----------------------------------------------
+    possible_layers <- emodnet_get_wfs_info(wfs)$layer_name
+    layers <- match.arg(layers, choices = possible_layers, several.ok = TRUE)
+
+    # check filter vector -----------------------------------------
+    cql_filter <- cql_filter %||% rep(NA, times = length(layers))
+    checkmate::assert_character(cql_filter, min.len = 1, any.missing = TRUE)
+
+    if (length(cql_filter) == 1 & length(layers) > 1) {
+        cql_filter <- rep(cql_filter, times = length(layers))
+        usethis::ui_info('{usethis::ui_field("cql_filter")} {usethis::ui_code(cql_filter)} recycled across all layers')
+    }
+
+    if (checkmate::test_named(cql_filter)) {
+        cql_filter <- cql_filter[layers]
     }
+    checkmate::assert_character(cql_filter, len = length(layers))
+
 
-        # get features
-    out <- purrr::map2(.x = layers, .y = cql_filter, ~ews_get_layer(.x, wfs, cql_filter = .y), wfs,
-                      suppress_warnings) %>%
-         stats::setNames(layers)
+    # get features -------------------------------------------------------------
+    out <- purrr::map2(
+        .x = layers, .y = cql_filter,
+        ~ews_get_layer(.x, wfs, cql_filter = .y),
+        wfs, suppress_warnings
+    ) %>%
+        stats::setNames(layers)
 
-    # if reduce_layers = T, reduce to single sf
-    if(reduce_layers){
+    # if reduce_layers = T, reduce to single sf --------------------------------
+    if (reduce_layers) {
         tryCatch(
             out <- purrr::reduce(out, rbind),
             error = function(e) {
                 usethis::ui_stop("Cannot reduce layers.
                              Try again with {usethis::ui_code('reduce_layers = FALSE')}")
             }
         )
-    }else{
-
     }
-    out <- standardise_crs(out, crs)
-    return(out)
+
+    standardise_crs(out, crs)
 }
 
 
 check_layer_crs <- function(layer_sf, layer, wfs) {
 
     sf_crs <- sf::st_crs(layer_sf)
+    if (!is.na(sf_crs) && !is.null(sf_crs)) {
```

Comment:
Oh, honest question. What is the reason this better than or? (`|`)

Reply:
Right, ok. and also we're getting out quicker i.e. fail early but also exit early if you can!